File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        for (int i = 0; i < prices.length; i++) {
            for (int j = 1 + i; j < prices.length; j++) {
                int curr = prices[j] - prices[i];
                max = Math.max(max, curr);
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total = 0
        left = 0
        for right in range(len(prices)):
            if prices[right] > prices[left]:
                total += prices[right] - prices[left]
                left += 1
        return total
            

"""
- Sliding Window Approach, keep left pointer and start it at 0, and then use
right pointer to iterate through all elements.
"""